_test/test_sqw/test_sqw_dnd_eval.m

```diff
@@ -0,0 +1,284 @@
+classdef test_sqw_dnd_eval < TestCase
+    % Series of tests to check the call to user functions always obey the
+    % same interface
+    
+    properties
+        data_dir
+        dnd_4_test
+        sqw_4_test
+    end
+    
+    methods
+        function obj=test_sqw_dnd_eval(name)
+            if nargin<1
+                name = 'test_sqw_dnd_eval';
+            end
+            obj = obj@TestCase(name);
+            obj.data_dir = fullfile(fileparts(fileparts(mfilename('fullpath'))),'test_sqw_file');
+            test_sqw_file = fullfile(obj.data_dir,'sqw_2d_2.sqw');
+            obj.sqw_4_test = read_sqw(test_sqw_file);
+            obj.dnd_4_test = read_dnd(test_sqw_file);
+        end
+        function test_tobyfit(obj)
+            %
+            sample=IX_sample(true,[1,0,0],[0,1,0],'cuboid',[0.04,0.03,0.02]);
+            obj.sqw_4_test = ...
+                set_instrument(obj.sqw_4_test, @maps_instrument,'-efix',800,'S');
+            obj.sqw_4_test = ...
+                set_sample(obj.sqw_4_test,sample);
+
+            
+            kk = tobyfit(obj.sqw_4_test);
+            kk = kk.set_fun (@obj.sqw_eval_tester);
+            kk = kk.set_pin(1);
+            kk = kk.set_bfun (@obj.funceval_tester2D);
+            kk = kk.set_bpin (1);
+            
+            err_message = '';
+            try
+                ds = kk.simulate;
+                %[my_fitted_data,a] = kk.fit; % do not fit-- will not
+                %converge
+                failed = false;
+            catch ME
+                failed = true;
+                err_message = ME.message;
+            end
+            assertFalse(failed,err_message);
+            
+            sig = ds.data.s;
+            pix = ds.data.pix;
+            assertEqual(pix.signal(2),numel(sig)+1);
+            assertEqual(size(pix,2)+numel(sig),pix.signal(1));
+            
+            assertEqual(sig(2),2);
+            assertElementsAlmostEqual(sig(1),403.0462,'absolute',0.0001);
+            
+        end
+        
+        
+        
+        function test_dispersion_sqw(obj)
+            %
+            err_message = '';
+            try
+                ds = dispersion(obj.sqw_4_test,@obj.sqw_disp_tester,[]);
+                %[my_fitted_data,a] = kk.fit; % do not fit-- will not
```

Comment:
similarly here

Reply:
I've tested it on every new type of function -- here it left loose, probably can be removed